### PR TITLE
test: fix the tests on windows + ci: test windows using GitHub Actions

### PR DESCRIPTION
**Pull request by aminya**
- This adds Windows to GitHub Actions. I have removed the Appveyor config file. You should stop its app in the repository settings.
- Fixes the tests on Windows by:
  - normalizing the usage files
  - using `temp_dir`

image

### CODE REVIEW

File: .gitattributes [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# line endings
+* text=auto eol=lf
+*.{cmd,[cC][mM][dD]} text eol=crlf
+*.{bat,[bB][aA][tT]} text eol=crlf
+*.{vcxproj,vcxproj.filters} text eol=crlf
```
Comment (jyn514) [inline .gitattributes:5]: Where do these come from? There aren't any cmd/bat files in the repo, are there?

File: tests/integration.rs [hunk 2]
```diff
@@ -251,7 +256,8 @@ fn golden_reference(args: &[&str], file: &str) -> TestResult {
     if std::env::var("BLESS").as_deref() == Ok("1") {
         fs::write(file, actual)?;
     } else {
-        let expected = fs::read_to_string(file).context("failed to read usage file")?;
+        let mut expected = fs::read_to_string(file).context("failed to read usage file")?;
+        content_modifier(&mut expected);
```
Comment (jyn514) [inline tests/integration.rs:260]: nit: please call `content_normalize` directly here, I don't think it needs to be customizable per-test.

**Review by jyn514 — APPROVED** [review]
This looks great, thank you! I have a very small nit on the tests, and I'll fix the clippy lints in a separate PR.

File: .gitattributes (hunk 1 above)
Comment (aminya) [reply .gitattributes:5]: That's the template I use by default everywhere without the need for manually checking the content of the repos.

File: tests/integration.rs (hunk 2 above)
Comment (aminya) [reply tests/integration.rs]: Fixed in [`355afec` (#79)](https://github.com/holmgr/cargo-sweep/pull/79/commits/355afecc621a96cefcd3c6c97dd71d26c8c5effc)

**Comment by aminya** [comment]
Thanks for the merge. 

Please also stop the Appveyor app from the settings. If you don't have access, there is a hacky way to suppress the error.
```
# empty appveyor
build: off

branches:
  only:
  - non-existing
```

**Comment by jyn514** [comment]
Hmm, I thought I disabled the appveyor build, but I still see it running on https://github.com/holmgr/cargo-sweep/pull/80: https://ci.appveyor.com/project/holmgr/cargo-sweep/builds/45770497

I guess it's only cloning the repo and doing nothing else, which is harmless if a little silly.

**Comment by jyn514** [comment]
ah, I needed to set it not to run on any branches as well - now it's working :)

Thanks for the help!

**Pull request merged**